lib/agent.js

```diff
@@ -250,33 +247,43 @@ Agent.prototype.start = function (opts) {
     return this
   }
 
-  this._origStackTraceLimit = Error.stackTraceLimit
-  Error.stackTraceLimit = 20 // require more than default 10 for `agentActivationMethodFromStartStack()`
+  // Log preamble showing agent, environment and config relevant info
+  const preambleData = this._conf.loggingPreambleData
+  const isPreviewVersion = version.indexOf('-') !== -1
   const startStack = {}
-  Error.captureStackTrace(startStack)
-  this._agentActivationMethod = agentActivationMethodFromStartStack(startStack, this.logger)
-
-  if (this._conf.logLevel === 'trace') {
-    // Attempt to load package.json from process.argv.
-    var pkg = null
-    try {
-      var basedir = path.dirname(process.argv[1] || '.')
-      pkg = require(path.join(basedir, 'package.json'))
-    } catch (e) {}
-
-    this.logger.trace({
-      pid: process.pid,
-      ppid: process.ppid,
-      arch: process.arch,
-      platform: process.platform,
-      node: process.version,
-      agent: version,
-      agentActivationMethod: this._agentActivationMethod,
-      startTrace: startStack.stack.split(/\n */).slice(1),
-      main: pkg ? pkg.main : '<could not determine>',
-      dependencies: pkg ? pkg.dependencies : '<could not determine>',
-      conf: this._conf.toJSON()
-    }, 'agent configured correctly')
+
+  if (this._conf.active && this._conf.serviceName) {
+    this._origStackTraceLimit = Error.stackTraceLimit
+    Error.stackTraceLimit = 20 // require more than default 10 for `agentActivationMethodFromStartStack()`
+    Error.captureStackTrace(startStack)
+    this._agentActivationMethod = agentActivationMethodFromStartStack(startStack, this.logger)
+    preambleData.activationMethod = this._agentActivationMethod
+
+    if (this._conf.logLevel === 'trace') {
+      // Attempt to load package.json from process.argv.
+      let pkg = null
+      try {
+        var basedir = path.dirname(process.argv[1] || '.')
+        pkg = require(path.join(basedir, 'package.json'))
+      } catch (e) {}
+
+      // Add stack & dependencies for extra information
+      preambleData.dependencies = pkg ? pkg.dependencies : '<could not determine>'
+      preambleData.startTrace = startStack.stack.split(/\n */).slice(1)
+    }
+  }
+
+  this.logger.info(preambleData, 'Elastic APM Node.js Agent, version: %s', version)
+
+  if (isPreviewVersion) {
+    this.logger.warn('Version %s is a pre-release and not intended for use in production environments', version)
+  }
+
+  // TODO: this deactivation should be handled as well when config resolution/normalizaiton process
```

Comment:
Hrm. I'm second guessing myself on whether it is the job of "lib/config/..." to do this. <shrug/> It could be in either place.